File: kyw0716/level0/연속된 수의 합.js
function getSumNumber(num) {
  let sum = 0;

  for (let i = 1; i < num; i++) {
    sum += i;
  }

  return sum;
}

function solution(num, total) {
  const sumNumber = getSumNumber(num);
  let keyNumber = (total - sumNumber) / num;
  let answer = [];

  for (let i = 0; i < num; i++) {
    answer.push(keyNumber + i);
  }

  return answer;
}

// https://school.programmers.co.kr/learn/courses/30/lessons/120923

File: sk000801/입문/연속된 수의 합.java
class Solution {
    public int[] solution(int num, int total) {
        int[] answer = new int[num];
        if(num%2==1) {
            for(int i = -num/2; i <= num/2; i++) {
                answer[num/2+i] = total/num+i;
            }
        } else {
            for(int i = 0; i < num; i++) {
                answer[i] = total/num-(num/2-1)+i;
            }
        }
        return answer;
    }
}